src/cargo/ops/cargo_compile.rs

```diff
@@ -62,6 +62,29 @@ pub struct CompileOptions<'a> {
     pub target_rustc_args: Option<&'a [String]>,
 }
 
+impl<'a> CompileOptions<'a> {
+    pub fn with_default<F, T>(config: &Config, mode: CompileMode, f: F) -> T
```

Comment:
It's kind of annoying with the borrows here - there needs to be a lifetime for `'a` (which is why this takes a closure, rather than returning a `CompileOptions` - I think the only way to do this would be to take a bunch of empty slices as args, which is pretty annoying. i don't think we can derive here.